src/currency-field/CurrencyField.ts

```diff
@@ -232,220 +292,223 @@ export class CurrencyField extends OmniFormElement {
+                    // Condition if the delete key is pressed on a desktop keyboard check if the entire value of the input is selected or a portion and update accordingly.
+                    else if (e.inputType === 'deleteContentForward') {
+                        if (input.value.length === input.selectionEnd && input.selectionStart !== input.selectionEnd) {
+                            let preNumericValue = '0';
+                            for (let index = 0; index < that.fractionalPrecision; index++) {
+                                preNumericValue += '0';
+                            }
+                            centValue = that._convertToCents(preNumericValue);
+                        } else if (input.selectionStart !== input.selectionEnd) {
+                            centValue = that._convertToCents(
+                                input.value.slice(0, input.selectionStart as number) + input.value.slice(input.selectionEnd as number)
+                            );
+                        }
+                    }
+                    const amountPart = centValue?.substring(0, centValue.length - that.fractionalPrecision) as string;
 
-        await this.updateComplete;
+                    const fractionPart = centValue?.slice(-that.fractionalPrecision);
 
-        // Length of the input value after the input value is updated and formatted.
-        const postValueLength = this._inputElement?.value.length;
+                    const parsedAmountPart = amountPart ? that._parseAmount(amountPart) : '0';
+                    input.value = parsedAmountPart + that.fractionalSeparator + fractionPart;
+                    const floatValue = that._formatToFloat(input.value);
+                    that.value = floatValue;
+                    that._dispatchCustomEvent(that.value as number);
 
-        // Position the caret in the correct position.
-        if ((postValueLength as number) > preValueLength) {
-            this._inputElement!.setSelectionRange(caretPosition! + 1, caretPosition! + 1);
-        } else if (preValueLength > postValueLength!) {
-            this._inputElement!.setSelectionRange(caretPosition! - 1, caretPosition! - 1);
-        } else {
-            this._inputElement!.setSelectionRange(caretPosition, caretPosition);
+                    input.selectionStart = input.selectionEnd = 10000;
+                    return;
+                }, 0);
+            }
+            //Ensuring on older devices that the caret doesn't jump when hitting the delete key on a virtual keyboard.
+            else {
+                // eslint-disable-next-line @typescript-eslint/no-this-alias
+                const that = this;
+
+                setTimeout(() => {
+                    input.value = that._formatToCurrencyValue('0');
+                    const floatValue = that._formatToFloat(input.value);
+                    that.value = floatValue;
+                    input.selectionStart = input.selectionEnd = 10000;
```

Comment:
Should this not be based on the input value length instead of a harcoded value? What if the input has more than 10000 characters?
```suggestion
                    input.selectionStart = input.selectionEnd = input.value?.length ?? 0;
```

Reply:
Implemented the suggested implementation